controller_test.py

```diff
@@ -116,25 +119,102 @@ def test_start_new_2player_game_calls_prepare_list_of_players_and_associated_fun
       "Please enter your name: "))
-    self.assertEqual(len(mock_builtin_input.call_args_list), 3)
+    self.assertGreaterEqual(len(mock_builtin_input.call_args_list), 3)
 
+  @patch('builtins.print')
   @patch('bites.Bites.__init__', return_value = None)
   @patch('bites.Bites.play_full_game')
-  @patch('builtins.input', return_value = 2)
+  @patch('builtins.input', side_effect = [2,'Mario', 'Luigi', 'top down'])
   def test_start_new_game_creates_instance_of_Bites_class(
-    self, mock_builtin_input, mock_bites_play, mock_bites_init):
+    self, mock_builtin_input, mock_bites_play, mock_bites_init, mock_builtin_print):
     # test 111
     start_new_game()
     self.assertEqual(mock_bites_init.call_count, 1)
 
+  @patch('builtins.print')
   @patch('bites.Bites.__init__', return_value = None)
   @patch('bites.Bites.play_full_game')
-  @patch('builtins.input', return_value = 2)
+  @patch('builtins.input', side_effect = [2,'Mario', 'Luigi', 'top down'])
   def test_start_new_game_calls_Bites_play_full_game_method(
-    self, mock_builtin_input, mock_bites_play, mock_bites_init):
+    self, mock_builtin_input, mock_bites_play, mock_bites_init, mock_builtin_print):
     # test 112
     start_new_game()
     mock_bites_play.assert_called_once()
 
+class ChooseAnthillRuleTest(unittest.TestCase):
+  @patch('builtins.input', return_value = "top down")
+  @patch('builtins.print')
+  def test_choose_anthill_rule_allows_choice_of_top_down_option(
+    self, mock_builtin_print, mock_builtin_input):
+    # test 137
+    expected_result = "top down"
+    self.assertEqual(choose_anthill_rule(), expected_result)
+
+  @patch('builtins.input', return_value = "bottom up")
+  @patch('builtins.print')
+  def test_choose_anthill_rule_allows_choice_of_bottom_up(
+    self, mock_builtin_print, mock_builtin_input):
+    # test 138
+    expected_result = "bottom up"
+    self.assertEqual(choose_anthill_rule(), expected_result)
+
+  @patch('builtins.input', return_value = "leave gaps")
+  @patch('builtins.print')
+  def test_choose_anthill_rule_allows_choice_of_leave_gaps(
+    self, mock_builtin_print, mock_builtin_input):
+    # test 139
+    expected_result = "leave gaps"
+    self.assertEqual(choose_anthill_rule(), expected_result)
+
+  @patch('builtins.input', side_effect = ["top up", "bottom up"])
+  @patch('builtins.print')
+  def test_user_makes_typo_no_error_redo_choice_select_botom_up(
+    self, mock_builtin_print, mock_builtin_input):
+    # test 140
+    expected_result = "bottom up"
+    self.assertEqual(choose_anthill_rule(), expected_result)
+
+  @patch('builtins.input', return_value = "random")
+  @patch('builtins.print')
+  def test_user_can_enter_random_and_one_of_the_other_options_is_selected(
+    self, mock_builtin_print, mock_builtin_input):
+    # test 143
+    expected_results = ["top down", "bottom up", "leave gaps"]
+    self.assertIn(choose_anthill_rule(), expected_results)
```

Comment:
This test could break if the `ANTHILL_CARD_DICT` is modified in the constants file. You should pass the `anthill_card_dict` argument into `choose_anthill_rules()` here in here. Otherwise consider importing the `ANTHILL_CARD_DICT` constant here; I realise this goes against my previous advice of not importing constants.

Reply:
Is it not OK for a test to fail when some part of the project is changed? If/when that situation occurs then would I not address the broken tests at tat point?

Reply:
okay